### ISSUE TITLE
YUM is deprecated

### ISSUE THREAD

**Issue by brejoc** [cobbler/cobbler#1980]
YUM is Python2 only and needs to be replaced with something else if repo mirroring is something cobbler should be able to do. @jmaas, what are your plans here?

**Comment by brejoc** [comment]
I was just taking an other look at the `api.py` and I'm wondering if this (`auto_add_repos` e.g.) is used at all. Then there is `action_reposync.py`, which guards the yum import, but still won't work for Py3.

**Comment by Conan-Kudo** [comment]
@brejoc It looks like this logic looks for either YUM or DNF. DNF is already packaged in openSUSE. If there's something missing from DNF that cobbler needs that you don't have, let me know, since I'm the DNF packager for openSUSE.

Let's just rip out all the YUM logic and assume DNF.

**Comment by Conan-Kudo** [comment]
We need to do this anyway since YUM is being removed in Fedora 30: https://fedoraproject.org/wiki/Changes/Retire_YUM_3

DNF is effectively YUM version 4.

**Comment by jmaas** [comment]
is DNF available in SLE ???

**Comment by Conan-Kudo** [comment]
It is currently available through SUSE PackageHub for SLE 15: https://packagehub.suse.com/packages/dnf/

As is the core plugins: https://packagehub.suse.com/packages/dnf-plugins-core/

If we need more, that can be arranged.

**Comment by summerm78** [comment]
What about CentOS 7 and RHEL? They still use yum.

**Comment by Conan-Kudo** [comment]
DNF was added to RHEL/CentOS 7 with the 7.6 release.

**Comment by opoplawski** [comment]
> Let's just rip out all the YUM logic and assume DNF.

If we are making any attempt to support EL7, please let's not.  yum is still the default there.

**Comment by Conan-Kudo** [comment]
@opoplawski Yes, but for the purpose of cobbler, that's really not important.

**Comment by Conan-Kudo** [comment]
@opoplawski Yes, but for the purpose of cobbler, that's really not important. One of the reasons DNF is now in EL7 is precisely so tools can move to it before YUM is gone completely. So we should take advantage of this boon and roll with it.

**Comment by JohanElmis** [comment]
RHEL and CentOS do use yum by default - as already mentioned.
It's also correct that 7.6 added dnf (but it's also Python2 interfaces it looks like (With C bindings).
Python3 isn't available by default - But it can be installed and run in a Python VirtualEnvironment.
RHEL 8 (that is soon to be released?) will still have yum (afaik) - but probably also dnf.
It will also have both Python2 and Python3 - which will be separate from the yum-version that is isolated so we don't risk to break something during an upgrade.

**Comment by JohanElmis** [comment]
Getting Cobbler 3 to run on a RHEL/CentOS 7 box will be somewhat complicated - so personally I'm fine with waiting to do this upgrade until 8 is out.

**Comment by Conan-Kudo** [comment]
@brejoc YUM has been removed from Fedora Rawhide. As DNF is available in all target Linux distributions, let's just switch everything over now.

**Comment by Conan-Kudo** [comment]
@brejoc, @jmaas, @JohanElmis So it looks like Python 3 is getting pulled into RHEL 7.7 (Cf. [RH#1639030](https://bugzilla.redhat.com/show_bug.cgi?id=1639030)). There's a separate issue requesting the Python 3 version of the DNF stack for RHEL 7 that you may want to put in feedback to voice its importance to you: [RH#1672102](https://bugzilla.redhat.com/show_bug.cgi?id=1672102).

**Comment by Conan-Kudo** [comment]
YUM has been removed from openSUSE Tumbleweed. The DNF stack now provides `yum` and `yum-utils` CLI tools, but the old Python API is completely gone.
